src/jltypes.c

```diff
@@ -2610,7 +2610,7 @@ void jl_init_types(void) JL_GC_DISABLED
                                                     "hash", "n_uninitialized",
                                                     "flags", // "abstract", "mutable", "mayinlinealloc",
                                                     "max_methods");
-    const static uint32_t typename_constfields[1] = { 0x00003a3f }; // (1<<0)|(1<<1)|(1<<2)|(1<<3)|(1<<4)|(1<<5)|(1<<9)|(1<<11)|(1<<12)|(1<<13)
+    const static uint32_t typename_constfields[1] = { 0x00003a27 }; // (1<<0)|(1<<1)|(1<<2)|(1<<5)|(1<<9)|(1<<11)|(1<<12)|(1<<13) ; TODO: put back (1<<3)|(1<<4) in this list
```

Comment:
If I'm understanding this correctly, this change resolves the bug where we were embedding a literal pointer constant directly into the IR. And the other effects annotations are introduced to preserve the availability of constant-folding, but in a more correct way—specifically by folding the reflection methods at their call sites. Is that accurate?

Reply:
Constant folding was broken before, due to the use of `unsafe_load`, but otherwise yes